### DISCUSSION TITLE
Using RBAC without authorization service

### DISCUSSION THREAD

**Discussion by fabio-percivaldi** [mia-platform/community#29] [Q&A] [answered]
I'm trying to use the RBAC container without having authorization-service in my project and i receive this error when i try to call a protected API
`
error: {
      "message": "user properties header is not valid: json: cannot unmarshal number into Go value of type map[string]interface {}"
    }
`
i'm guessing that since no one is configuring the user properties correctly in the header (usually it is the authorization-service) the marshaling fails. 

I'm not using the authorization service since we don't have auth0 o client credentials so we can't configure a /user-info for the authorization service, and we don't actually need it. 

Am i missing something or RBAC requires the authorization-service to work?

**Comment by fredmaggiowski** [comment, marked as answer]
This setup is something we would like to do us too, so it is something we'd like to work on in the future. 

In the meantime we could add a feature to Rönd in order to ignore missing headers and prevent this error to appear, I'd suggest you to open a issue or a discussion on the [Rönd](https://github.com/rond-authz/rond/) repository.

**Reply by fabio-percivaldi** [reply]
i opened a new issue on the rond project https://github.com/rond-authz/rond/issues/69

**Comment by davidebianchi** [comment]
Yes, I suppose this should be fixed. Can you tell us what is the use case to use RBAC with rond if you don't have logged users?

**Reply by fabio-percivaldi** [reply]
i'm doing a training where i would like to show a very easy allow policy that checks a query string, in a real scenario this does not make any sense, but i think there could be a case where we want to apply a check on some input property (like body, query string or headers) even if the user is not logged in, don't we?

**Reply by davidebianchi** [reply]
Yeah, sure. It's an authorization framework, so the only case I thought about is an API which responds in different way (so, with a specific filter) if the request is public or it is from a logged user. I was curious to know if you have other use cases. This could be interesting to report as use case also in Rond. I add it to the issue you open

**Comment by davidebianchi** [comment]
As a workaround, meanwhile the issue in Rond is not solved, you can go to the extensions section of the Console, under the api-gateway service you can modify the `auth-usage.conf` file adding:

```
set $mia_userid "";
set $mia_groups "unlogged";
set $mia_allowed "1";
set $mia_userproperties "{}";
```

⚠️ This configuration will remain also if you add an authorization service. So, if you want to enable the authorization you MUST remove it from the extensions. I suggest it to you only because you said it's only for test purposes.

**Reply by fredmaggiowski** [reply]
An empty string should be enough as well: https://github.com/rond-authz/rond/pull/70
